client/src/setup/setupIndexingHooks.ts

```diff
@@ -15,34 +13,14 @@ export function setupIndexingHooks(
   extensionState: ExtensionState,
   client: LanguageClient
 ): void {
+  // Show spinner when starting initialization
+  const indexingStatusItem = setupIndexingLanguageStatusItem();
+
   client
     .onReady()
     .then(() => {
-      const indexingStartDisposable = client.onNotification(
-        "custom/indexing-start",
-        () => {
-          extensionState.indexingFinished = false;
-
-          const indexingStatusItem = setupIndexingLanguageStatusItem();
-
-          extensionState.currentIndexingJobs.push(indexingStatusItem);
-        }
-      );
-
-      const indexingEndDisposable = client.onNotification(
-        "custom/indexing-end",
-        () => {
-          extensionState.indexingFinished = true;
-
-          const indexingStatusItem = extensionState.currentIndexingJobs.find(
-            (statusItem) => statusItem.id === INDEXING_JOB_ID
-          );
-
-          indexingStatusItem?.dispose();
-
-          triggerValidationForOpenDoc(client);
-        }
-      );
+      // Hide spinner when initialization ends
+      indexingStatusItem.dispose();
 
       const fileIndexedDisposable = client.onNotification(
```

Comment:
What do you think would be involved in getting rid of this?

Reply:
Is this the best way of approaching this though? I don't mean for this PR.

I am just interested in whether there is a better way of getting the language status item set per file. Is it possible just to set it on the open editor event through a custom call to the language server?